tfcoreml/_layers.py

```diff
@@ -982,16 +982,37 @@ def transpose(op, context):
   assert len(axes) == 4, "Op Transpose conversion only works with 4D tensors"
 
   # TODO - only works for 4D tensor without batch axis
-  target_batch_idx = axes.index(0) # the assumed TF batch axis
-  target_height_idx = axes.index(1) # the assumed TF height axis
-  target_width_idx = axes.index(2) # the assumed TF width axis
-  target_channel_idx = axes.index(3) # the assumed TF channel axis
-
-  coreml_axes = [0]*4
-  coreml_axes[target_batch_idx] = 0
-  coreml_axes[target_height_idx] = 2
-  coreml_axes[target_width_idx] = 3
-  coreml_axes[target_channel_idx] = 1
+  assert axes[0] == 0, "only works for 4D tensor without batch axis"
+
+  #Permutation without swapping appears to give wrong results, which may be
+  #a bug in coreml itself
+  if axes[1] != 1 and axes[2] != 2 and axes[3] != 3:
```

Comment:
that means you are checking for axes = [0,1,2,3], in which case the permute is doing nothing?

Reply:
It's "1=", rather than "==", which catches the "bad" permutes [0, 2, 3, 1] and [0, 3, 1, 2].